### PR TITLE
convert ConfirmAssetViewController

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

convert `ConfirmAssetViewController` to Swift

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -25,6 +25,6 @@ github "wireapp/wire-ios-share-engine" "170.0.0"
 github "wireapp/wire-ios-sync-engine" "302.0.2"
 github "wireapp/wire-ios-system" "28.1.0"
 github "wireapp/wire-ios-testing" "20.1.0"
-github "wireapp/wire-ios-transport" "66.2.0"
+github "wireapp/wire-ios-transport" "66.3.0"
```
Comment (johnxnguyen) [inline Cartfile.resolved]: Looks like an unrelated commit

File: Wire-iOS/Sources/Authentication/Landing/LandingViewController.swift [hunk 2]
```diff
@@ -351,7 +351,7 @@ final class LandingViewController: AuthenticationStepViewController {
     
     var isCustomBackend: Bool {
         switch BackendEnvironment.shared.environmentType.value {
-        case .production, .staging, .qaDemo:
+        case .production, .staging, .qaDemo, .qaDemo2:
             return false
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Authentication/Landing/LandingViewController.swift]: What's `qaDemo2` for?

File: Cartfile.resolved (hunk 1 above)
Comment (billypchan) [reply Cartfile.resolved]: fixed branch merging

File: Wire-iOS/Sources/UserInterface/Components/Controllers/ImagePickerConfirmationController.swift [hunk 3]
```diff
@@ -56,23 +56,21 @@ extension ImagePickerConfirmationController: UIImagePickerControllerDelegate {
...
+            let context = ConfirmAssetViewController.Context(asset: .image(mediaAsset: image), onConfirm: { [weak self] editedImage in
                 if let editedImage = editedImage {
                     self?.imagePickedBlock?(editedImage.pngData())
                 } else {
                     self?.imagePickedBlock?(image.pngData())
                 }
-            }
+                },
+                                                             onCancel: {
+                    picker.dismiss(animated: true)
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Components/Controllers/ImagePickerConfirmationController.swift]: Can you find a way to format these (and all the other) blocks better? Perhaps passing them in as arguments?

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Components/Controllers/ImagePickerConfirmationController.swift]: fixed
